### PR TITLE
prevent publish when package is private

### PR DESCRIPTION
**Pull request by EndangeredMassa**
addresses https://github.com/EndangeredMassa/npub/issues/20

### CODE REVIEW

**Comment by EndangeredMassa** [comment]
I tested this with a dummy module and it worked as expected.

File: src/publish/index.coffee [hunk 1]
```diff
@@ -33,6 +34,10 @@ module.exports = (dir, log, config, version, testCommand) ->
   npm = Npm(dir, log)
   changelog = Changelog(dir, git)
 
+  if isPrivate(dir)
+    console.log 'Cannot publish this package because it is private.'
+    return
```
Comment (khoomeister) [inline src/publish/index.coffee]: could exit with non-zero code?

Comment (EndangeredMassa) [inline src/publish/index.coffee]: sounds like a good idea

**Comment by khoomeister** [comment]
Aside from that, LGTM

**Comment by EndangeredMassa** [comment]
Now exits with code 2 when package is private

**Comment by khoomeister** [comment]
Nice - LGTM

**Comment by EndangeredMassa** [comment]
published in [1.0.0](https://github.com/EndangeredMassa/npub/releases/tag/v1.0.0)

**Pull request merged**